content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md

```diff
@@ -0,0 +1,317 @@
+
+                  name: mysql-user-pass
+
+                  key: password
+
+            ports: 
+
+            - containerPort: 3306        # The port that the container exposes        
+
+              name: mysql
+
+            volumeMounts: 
+
+            - name: mysql-persistent-storage  # This name should match the name specified in `volumes.name` 
+
+              mountPath: /var/lib/mysql
+
+          volumes:                       # A PersistentVolume is mounted as a volume to the Pod   
+
+          - name: mysql-persistent-storage
+
+            persistentVolumeClaim: 
+
+              claimName: mysql-pv-claim
+
+In short, we are creating the following four resources in the above
+manifest file: 
+
+-   PV
+-   A PVC for requesting access to the PV resource
+-   A service for having a static endpoint for the MySQL database
+-   A deployment for running and managing the MySQL Pod
+
+The MySQL container reads database credentials from environment
+variables. The environment variables access these credentials from
+Kubernetes secrets.\
+ For the next step, you can start a Minikube cluster to create
+Kubernetes secrets to store database credentials and deploy the MySQL
+instance.
+
+### Summary
+
+In this article, you learn about how to deploy MySQL on
+[Kubernetes](https://myhsts.org/tutorial-review-of-17-essential-topics-for-mastering-kubernetes.php)
+and in doing so you learned three important Kubernetes concepts:
+Persistent Volume, Persistent Volume Claim and Secrets. We also reviewed
+a sample configuration environment for setting the right properties for
+PV, PVC and Secrets on the manifest file. You can extend this article by
+using Kubernetes tools like Minikube to make Kubernetes secrets to store
+database credentials and deploy the MySQL instance.
+
+### Resources
+
+For more discussions and helps on Kubernetes, reading the following
+articles is highly recommended:
+
+-   [17 Best Practices for Managing Kubernetes
+    Containers](https://myhsts.org/tutorial-review-of-17-essential-topics-for-mastering-kubernetes.php)
+-   [Advance System Admin Guide- 9 Best Practices for Managing
+    Kubernetes](https://blockchain.dcwebmakers.com/blog/advance-topics-for-deploying-and-managing-kubernetes-containers.html)
+-   [System Admin Guide- What is Kubernetes and how it
+    works](https://coding-bootcamps.com/blog/kubernetes-evolution-from-virtual-servers-and-kubernetes-architecture.html)
+-   [Comprehensive Guide for Migration From Monolithic To Microservices
+    Architecture](https://blockchain.dcwebmakers.com/blog/comprehensive-guide-for-migration-from-monolithic-to-microservices-architecture.html)
+-   [Review of Pod-to-Pod Communications in
+    Kubernetes](https://coding-bootcamps.com/blog/pod-to-pod-communications-in-kubernetes.html)
+-   [Build and Deploy Containerized Applications with Golang on
+    Kubernetes](https://coding-bootcamps.com/blog/build-containerized-applications-with-golang-on-kubernetes.html)
```

Comment:
I would omit these links in favor of internal hyperlinks.

Reply:
I removed all of them